### DISCUSSION TITLE
关于增加权重分值计算排序的讨论

### DISCUSSION THREAD

**Discussion by LiLittleCat** [LiLittleCat/awesome-free-chatgpt#80] [Ideas]
本项目旨在收集开箱即用的免费镜像站，目前收录的站点较多，可能需要寻找很久才能找到合适的站点。

相关 issue: #76 #79

拟根据站点属性计算站点得分，并按照得分进行排序，属性和分值如下：

| 属性                      | 分值    |
|-------------------------|-------|
| 👍维护者推荐                 | 5     |
| 🆓完全免费，打开即用             | 2     |
| 🔓有免费额度                 | 1     |
| 🔒需要登陆                  | -5    |
| 🔑需要输入 API Key          | -3    |
| 💰需要充值                  | -5    |
| 👀需要关注公众号               | -5    |
| 💪支持 GPT4               | 1     |
| 🧰不仅仅是 Chat，支持其他模型或其他功能 | 0.5   |
| 🌎需要国际网络                | -1    |
| 🌟社区推荐（issue 形式）        | 0.5/人 |

**Comment by yesaiwen2008** [comment]
这个权重排序的创意不错。 
小建议：对于调用bing的网站，可否做特别标注？
比如说：https://dev.yqcloud.top/    它的使用体验非常好。
但有时候想找ChatGPT的时候，会误点击它。 
如果能有特殊标记，体验会更好。

**Reply by LiLittleCat** [reply]
怎么知道他是调用 bing 的？

**Reply by yesaiwen2008** [reply]
通过搜索结果，猜测的。

例如：同一个问题，GitHub 更新fork的仓库

**ChatGPT 3.5的回答：**
`要将 fork 的仓库与原始仓库同步更新，可以按照以下步骤进行操作：

首先，在本地将 fork 的仓库克隆到计算机上。

使用以下命令将原始仓库添加为远程仓库：

git remote add upstream [原始仓库的 URL]
确认远程仓库是否成功添加：

git remote -v
从原始仓库拉取最新更改：`

**新bing的回答**
正在搜索: GitHub 更新fork的仓库

正在搜索: GitHub 更新fork的仓库

正在为你生成答案…

正在为你生成答案…

您可以通过以下步骤更新fork的仓库：

在GitHub上进入fork的项目。
点击右边的pull request。
点击下面的switching the base。
点击“Create pull request”即可。
[如果您想了解更多细节，可以参考这些链接](https://beltxman.com/3320.html)

**dev.yqcloud.top的回答：**
正在为您搜索：GitHub、更新、fork、仓库

如果你 fork 了其他人的仓库，想要将其更新到原仓库的最新版本，可以按照以下步骤进行：

Clone 你 fork 的仓库到本地。
添加一个指向原仓库的 upstream 远程仓库：git remote add upstream [原仓库链接]。例如： git remote add upstream https://github.com/ORIGINAL_OWNER/ORIGINAL_REPOSITORY.git
拉取 upstream 远程仓库的更新：git fetch upstream。
切换到你的本地 master 分支：git checkout master。
合并 upstream 的更新到你的本地仓库：git merge upstream/master。
将更新推送到你的 GitHub 仓库：git push origin master。
参考来源：[[1](https://github.com/)]。

**Reply by LiLittleCat** [reply]
他那个发送旁边有个联网支持，应该是这个开关

**Comment by RGyAIjhaole** [comment]
笑死

**Comment by NianBroken** [comment]
@LiLittleCat 我认为“需要登陆”不应该扣那么多的分。如果登录后是免费使用，那么他与“完全免费”的区别并不大。
另外，如果它持续免费的时间较长，即从“添加时间”起就开始免费，直到现在依然免费，这个免费时间越长，就可以加更多的分。

**Reply by zsh2401** [reply]
非常同意…

**Comment by LiLittleCat** [comment]
初步构思了一下优化方案，括号内为分值：
1. 登录层面

- 无需登录（10）
- 需要登录（包括注册账号和微信扫码、关注公众号等）（5）

2. 模型支持
   支持的模型分数累加，模型分类列表需完善
  - 普通模型（1），如 GPT 4o mini
- 高级模型（2），如 GPT 4o，Claude 3.5 Sonnet
- 顶级模型（4），如 o1-mini，o1-preview，o1

3. 付费情况
   支持的模型分数\*倍率，越高级的模型越免费，得分越高

- 免费模型（\*10）
- 额度模型（\*5）
- 付费模型（\*1）

4. 其他功能

- 支持绘图（5）
- 支持联网搜索（5）
- more...

5. 存活时间
从添加时间开始计算，每正常运行一天累计 0.1 分

@NianBroken @zsh2401 可以交流下，提下意见😊
